lib/k8s-e2e-tests/src/lib.rs

```diff
@@ -1,32 +1,69 @@
+use indoc::indoc;
 use k8s_openapi::{
-    api::core::v1::{Container, Pod, PodSpec},
-    apimachinery::pkg::apis::meta::v1::ObjectMeta,
+    api::core::v1::{Affinity, Container, Pod, PodAffinity, PodAffinityTerm, PodSpec},
+    apimachinery::pkg::apis::meta::v1::{LabelSelector, ObjectMeta},
 };
-use k8s_test_framework::{Framework, Interface, Reader};
+use k8s_test_framework::{
+    test_pod, wait_for_resource::WaitFor, CommandBuilder, Framework, Interface, Manager, Reader,
+};
+use std::collections::BTreeMap;
 use std::env;
+use tracing::{debug, error, info};
 
 pub mod metrics;
 
 pub const BUSYBOX_IMAGE: &str = "busybox:1.28";
 
+pub fn init() {
+    let _ = env_logger::builder().is_test(true).try_init();
+}
+
 pub fn get_namespace() -> String {
-    env::var("NAMESPACE").unwrap_or_else(|_| "test-vector".to_string())
+    use rand::Rng;
+
+    // Generate a random alphanumeric (lowercase) string to ensure each test is run with unique
+    // names.
+    // There is a 36 ^ 5 chance of a name collision, which is likely to be an acceptable risk.
+    let id: String = rand::thread_rng()
+        .sample_iter(&rand::distributions::Alphanumeric)
+        .take(5)
+        .map(|num| (num as char).to_ascii_lowercase())
+        .collect();
+
+    format!("test-vector-{}", id)
 }
 
-pub fn get_namespace_appended(suffix: &str) -> String {
-    format!("{}-{}", get_namespace(), suffix)
+pub fn get_namespace_appended(namespace: &str, suffix: &str) -> String {
+    format!("{}-{}", namespace, suffix)
 }
 
 /// Gets a name we can use for roles to prevent them conflicting with other tests.
 /// Uses the provided namespace as the root.
-pub fn get_override_name(suffix: &str) -> String {
-    format!("{}-{}", get_namespace(), suffix)
+pub fn get_override_name(namespace: &str, suffix: &str) -> String {
+    format!("{}-{}", namespace, suffix)
 }
 
 /// Adds a fullnameOverride entry to the given config. This allows multiple tests
 /// to be run against the same cluster without the role anmes clashing.
 pub fn config_override_name(config: &str, name: &str) -> String {
-    format!("fullnameOverride: \"{}\"\n{}", name, config)
+    if env::var("multinode".to_string()) == Ok("yes".to_string()) {
```

Comment:
nit: I'd just accept any value for `MULTINODE` (I'd also uppercase it). Then just check for `is_ok()` in this condition. I think it's fairly common practice to consider "var is set" to mean "true" and non-set values to mean false.